oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/base/StorageEsInstaller.java

```diff
@@ -68,26 +94,39 @@ protected boolean isExists(Model model) throws StorageException {
     @Override
     protected void createTable(Model model) throws StorageException {
         ElasticSearchClient esClient = (ElasticSearchClient) client;
-
         Map<String, Object> settings = createSetting(model);
         Map<String, Object> mapping = createMapping(model);
-        log.info("index {}'s columnTypeEsMapping builder str: {}", esClient.formatIndexName(model.getName()), mapping
-            .toString());
+        String tableName = StoragePartitioner.INSTANCE.getTableName(model);
+        StoragePartitioner.INSTANCE.registerTableRelation(model.getName(), tableName);
+        log.info("index {}'s columnTypeEsMapping builder str: {}",
+                 esClient.formatIndexName(tableName), mapping.toString()
+        );
 
+        String indexName;
         try {
-            String indexName;
-            if (!model.isTimeSeries()) {
-                indexName = model.getName();
-            } else {
-                if (!esClient.isExistsTemplate(model.getName())) {
-                    boolean isAcknowledged = esClient.createTemplate(model.getName(), settings, mapping);
-                    log.info(
-                        "create {} index template finished, isAcknowledged: {}", model.getName(), isAcknowledged);
+            if (model.isTimeSeries()) {
+                Map<String, Object> template = esClient.getTemplate(tableName);
+                if (!template.isEmpty()) {
+                    appendTemplateMapping(tableName, (Map<String, Object>) template.get("mappings"));
+                }
+                if (!containsTemplateMapping(tableName, mapping)) {
+                    Map<String, Object> templateMapping = appendTemplateMapping(tableName, mapping);
+                    boolean isAcknowledged = esClient.createTemplate(tableName, settings, templateMapping);
```

Comment:
Are you meaning we could recreate a template w/o deleting the old one?

Reply:
You need to polish this new creating logic, it is not clear to me.

Reply:
create template using PUT method, so don't need to delete

Reply:
Then let's polish the logic like this
1. Check template existing. 
2. If no template, creating.
3. If yes, check mapping compatible, then merging and update(put).
4. Create latest index.